python/ray/train/_internal/storage.py

```diff
@@ -614,37 +592,92 @@ def experiment_fs_path(self) -> str:
 
-        This local "cache" path refers to location where files are dumped before
-        syncing them to the `storage_path` on the `storage_filesystem`.
+        The directory is synced to
+        `{storage_path}/{experiment_dir_name}` periodically.
+        See `_ExperimentCheckpointManager.checkpoint` for where that happens.
         """
-        return Path(self.storage_local_path, self.experiment_dir_name).as_posix()
+        return Path(
+            _get_ray_train_session_dir(),
+            self._timestamp,
+            self.experiment_dir_name,
+            "driver_artifacts",
+        ).as_posix()
 
     @property
-    def trial_local_path(self) -> str:
-        """The local filesystem path to the trial directory.
+    def trial_fs_path(self) -> str:
+        """The trial directory path on the `storage_filesystem`.
 
         Raises a ValueError if `trial_dir_name` is not set beforehand.
         """
         if self.trial_dir_name is None:
             raise RuntimeError(
-                "Should not access `trial_local_path` without setting `trial_dir_name`"
+                "Should not access `trial_fs_path` without setting `trial_dir_name`"
+            )
+        return Path(self.experiment_fs_path, self.trial_dir_name).as_posix()
+
+    @property
+    def trial_local_staging_path(self) -> str:
+        """The local filesystem path of the trial directory on the driver.
+
+        The driver is the node where `Trainer.fit`/`Tuner.fit` is being called.
+
+        This path is of the form:
+        `/tmp/ray/session_<session_id>/artifacts/<ray-train-job-timestamp>/
+        <experiment_dir_name>/driver_artifacts/<trial_dir_name>`
+
+        This should be used as the temporary location for files on the driver
+        before syncing them to `(storage_filesystem, storage_path)`.
+
+        For example, callbacks (e.g., JsonLoggerCallback) should write trial-specific
+        logfiles within this directory.
+        """
+        if self.trial_dir_name is None:
+            raise RuntimeError(
+                "Should not access `trial_local_staging_path` "
+                "without setting `trial_dir_name`"
             )
-        return Path(self.experiment_local_path, self.trial_dir_name).as_posix()
+        return Path(self.experiment_local_staging_path, self.trial_dir_name).as_posix()
 
     @property
-    def trial_fs_path(self) -> str:
-        """The trial directory path on the `storage_filesystem`.
+    def trial_working_directory(self) -> str:
+        """The local filesystem path to trial working directory.
 
-        Raises a ValueError if `trial_dir_name` is not set beforehand.
+        This path is of the form:
+        `/tmp/ray/session_<session_id>/artifacts/<ray-train-job-timestamp>/
+        <experiment_dir_name>/working_dirs/<trial_dir_name>`
+
+        Ray Train/Tune moves the remote actor's working directory to this path
+        by default, unless disabled by `RAY_CHDIR_TO_TRIAL_DIR` environment variable.
```

Comment:
What would happen if `RAY_CHDIR_TO_TRIAL_DIR=0`? I saw we are always using  `_get_ray_train_session_dir()`.

Reply:
The remote actors will just not be moved to this directory. However, this is still the directory that would be synced to storage if `sync_artifacts=True`. So, users would need to write files to `storage.trial_working_directory` if they wanted to use that feature.

I need to upload this faq section: https://docs.ray.io/en/latest/tune/faq.html#how-do-i-access-relative-filepaths-in-my-tune-training-function